components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7436,6 +7436,34 @@ private void handleException(String msg, Throwable t) throws APIManagementExcept
         return uriTemplates;
     }
 
+    public Boolean isAPIProductized(String uuid) throws APIManagementException {
+        Map<Integer, URITemplate> uriTemplatesOfAPI = getURITemplatesForAPI(uuid);
+        try {
+            return isUriTemplateAssociatedWithAPIProduct(uuid, uriTemplatesOfAPI);
+        } catch (SQLException e) {
+            handleException("Failed to get API Products associate with the URI templates of API " + uuid, e);
```

Comment:
getURITemplatesForAPI method throws APIManagementException. Why do we have to handle SQLException here?